### ISSUE TITLE
Including external JavaScript that you can't just copy

### ISSUE THREAD

**Issue by PragTob** [activeadmin/activeadmin#5170]
#5060 deprecated including javascripts in favor of using the asset pipeline. While good in spirit, this makes the world for external javascripts that you can't just copy locally a lot harder [as pointed out](https://github.com/activeadmin/activeadmin/pull/5060#issuecomment-325805074).

Would be great if we had an easy (tm) way to also include these sorts of javascript resources (like google maps etc.).

Thanks for all the work!
Tobi

**Comment by phillc** [comment]
Or make customizing the template easier, which would allow us to just insert a script tag, but I figured undoing the recent change would be easier.

**Comment by varyonic** [comment]
@javierjulio Maybe related, do you have a better answer for [this question on StackOverflow?](https://stackoverflow.com/questions/44840109/can-activeadmin-load-javascript-bundled-by-webpack-rails-5-1/)

**Comment by javierjulio** [comment]
@varyonic definitely related. But to stay focused on your question, I don't have a better answer for now, sorry. 😞

Now in regards to this issue, I do believe that the solution to this is that users should be able to install the layout and it should be as a traditional Rails layout in ERB, not be in ARB. I understand why others need to bring in JS files and Rails gives us that solution already: layouts. I don't think providing a DSL for doing that is right. We'd have to do that for a lot of other cases. I was going to bring up meta tags and then I found out we already have config items for that. 🙄😂 I think ActiveAdmin should have a  layout with sensible defaults but with that, users can customize as needed rather than us providing every possible option/config/etc.

I'm struggling to find time for this but this is something we should do. I see it very much like how Devise or Kaminari has their template files while also using methods they have implemented.

**Comment by varyonic** [comment]
Like [activeadmin_custom_layout](https://github.com/bys-control/activeadmin_custom_layout) maybe?

**Comment by javierjulio** [comment]
@varyonic yes! I didn't think to look for a gem. Great find! I think we should be looking at that to integrate into AA whether completely or some variation of that. I was going to start on this but from scratch, so thanks! That gem can give us a good starting point.

**Comment by PragTob** [comment]
Giving us a possibility to extend a custom layout or just override is definitely better and solves many more problems. Thank you!

**Comment by PragTob** [comment]
In case anyone comes here wondering, the "freedom patch" to get rid off the warning is rather simple:

```
module ActiveAdmin
  class Application
    def register_javascript(name)
      javascripts.add name
    end
  end
end
```

A method to customize the layout would still be great but I know y'all got a lot on your plate :)

Thaks!

**Comment by xn** [comment]
@PragTob's suggestion no longer works after #5142

**Comment by deivid-rodriguez** [comment]
Hei, so... This is essentially the only thing bothering me about releasing 2.0.0. Is our official solution to this problem recommending this external gem, or are we planning to provide something built-in to customize the layout?

My proposal for now is to revert this deprecation, release what we have in master, and then see what we do about this for ActiveAdmin 3.

Thoughts?

**Comment by varyonic** [comment]
I studied activeadmin_custom_layout and I can't recommend it. I did start on [changing ActiveAdmin to use Rails layouts](https://github.com/varyonic/activeadmin/commits/layout-refactor) but it's no longer a priority for me.

**Comment by deivid-rodriguez** [comment]
Thanks @varyonic for your input, and also for sharing your initial work on this. :+1:

I think this could be a good move, but we need better alternatives to propose to users. Since my focus is to release all the frontend improvements we have in master, I plan to revert this for the time being to unblock the rest of the improvements, unless another @activeadmin/maintainers have different opinions.

**Comment by PragTob** [comment]
I think rolling it back is the most pragmatic thing to do

**Comment by rdj** [comment]
In case anybody else finds this like I do, trying to work around the deprecation warning on an older version, in lieu of the "freedom patch" recommended above, consider just silencing the deprecation:
```
  ActiveSupport::Deprecation.silence do
    config.register_javascript '//www.google.com/jsapi'
  end
```
